digiwf-integrations/digiwf-ticket-integration/digiwf-ticket-integration-core/src/main/java/de/muenchen/oss/digiwf/ticket/integration/adapter/out/s3/S3Adapter.java

```diff
@@ -0,0 +1,95 @@
+
+    @Override
+    public List<FileContent> loadFiles(final List<String> filepaths, final String processDefinition) {
+        final String s3Storage = getDomainSpecificS3Storage(processDefinition).orElse(null);
+        final List<FileContent> contents = new ArrayList<>();
+        filepaths.forEach(path -> {
+            if (path.endsWith("/")) {
+                contents.addAll(getFilesFromFolder(path, s3Storage));
+            } else {
+                contents.add(getFile(path, s3Storage));
+            }
+        });
+        return contents;
+    }
+
+    private List<FileContent> getFilesFromFolder(String folderpath, final String domainSpecificS3Storage) {
+        try {
+            final List<FileContent> contents = new ArrayList<>();
+            final Set<String> filepath = documentStorageFolderRepository.getAllFilesInFolderRecursively(folderpath, domainSpecificS3Storage).block();
+            filepath.forEach(file -> contents.add(getFile(file, domainSpecificS3Storage)));
+            return contents;
+        } catch (final DocumentStorageException | DocumentStorageServerErrorException |
+                       DocumentStorageClientErrorException e) {
+            throw new BpmnError("LOAD_FOLDER_FAILED", "An folder could not be loaded from url: " + folderpath);
+        }
+    }
+
+    private FileContent getFile(String filepath, final String domainSpecificS3Storage) {
+        try {
+            final Tika tika = new Tika();
+            final byte[] bytes = this.documentStorageFileRepository.getFile(filepath, 3, domainSpecificS3Storage);
+            final String mimeType = tika.detect(bytes);
+            final String filename = FilenameUtils.getName(filepath);
+
+            // check if mimeType exists
+            supportedExtensions
+                    .stream()
+                    .filter(extension -> extension.equals(mimeType))
+                    .findAny()
+                    .orElseThrow(() -> new BpmnError("FILE_TYPE_NOT_SUPPORTED", "The type of this file is not supported: " + filepath));
+
+            return new FileContent(mimeType, filename, bytes);
+        } catch (final DocumentStorageException | DocumentStorageServerErrorException |
+                       DocumentStorageClientErrorException e) {
+            throw new BpmnError("LOAD_FILE_FAILED", "An file could not be loaded from url: " + filepath);
+        }
+    }
+
+    private Optional<String> getDomainSpecificS3Storage(final String processDefinition) {
+        try {
+            final ProcessConfigTO processConfig = processConfigApi.getProcessConfig(processDefinition);
+            return processConfig.getConfigs().stream()
+                    .filter(cfg -> cfg.getKey().equals(APP_FILE_S3_SYNC_CONFIG))
+                    .findAny()
+                    .map(ConfigEntryTO::getValue);
+        } catch (final Exception e) {
+            return Optional.empty();
+        }
+
+    }
```

Comment:
Und das hätte man eigntl auch gleich in die S3-Lib einbauen können oder meinst du das ist mehr Aufwand?

Reply:
Gleicher Punkt wie oben. Das sollten wir eher im Rahmen von https://github.com/it-at-m/digiwf-core/issues/1442 machen.